### PR TITLE
Add Bench Config Params to Result and Warm Up Phase

### PR DESCRIPTION
**Pull request by brunoborges**
This PR does the following:

1. Builds a Docker image for `ghz` with a version we can control ourselves.
1. Reports the configuration parameters used for the benchmark.
1. Adds a warm-up phase of `30s` that won't collect data.
1. Makes all env vars available to all subsequent scripts by exporting them.
1. Pass the GRPC_SERVER_RAM to the server container

### CODE REVIEW

File: analyze.sh [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
 echo "-----"
 echo "Benchmark finished. Detailed results are located in: ${RESULTS_DIR}"
+
 docker run --name analyzer --rm \
 	-v "${PWD}/analyze:/analyze:ro" \
 	-v "${PWD}/${RESULTS_DIR}:/reports:ro" \
 	ruby:2.7-buster ruby /analyze/results_analyze.rb reports ||
 	exit 1
+
+cat > ${RESULTS_DIR}/bench.params << EOF
```
Comment (LesnyRumcajs) [inline analyze.sh:13]: We could also add the current commit hash, it's somehow also a quite important factor

**Comment by LesnyRumcajs** [comment]
I think this PR may be missing a directory. :)

**Comment by Trisfald** [comment]
What is the advantage to build our own `ghz` image? Would it be useful to make it available on docker hub?

**Comment by brunoborges** [comment]
Done, `ghz-tool` folder added. Totally missed to add the directory, indeed :-)

@Trisfald, the infobox image runs an older version. That's all. And I don't see a reason to publish either. But we could propose to the ghz project to publish.

File: build.sh [hunk 2]
```diff
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+# Build ghz Docker image.
+# See ghz-tool/Dockerfile for details/version
+docker build -t ghz_bench:latest ./ghz-tool/
```
Comment (Trisfald) [inline build.sh:5]: We should update the cleanup script at line 6 with the `ghz` image name

Comment (brunoborges) [reply build.sh:5]: Fixed on 196e4b6.

File: README.md [hunk 3]
```diff
@@ -44,6 +44,7 @@ The benchmark can be configured through the following environment variables:
 |**Name**|**Description**|**Default value**|
 |--------|---------------|:---------------:|
 |GRPC_BENCHMARK_DURATION|Duration of the benchmark.|30s|
+|GRPC_BENCHMARK_WARMUP|Duration of the warmup. Stats won't be collected.|30s|
```
Comment (Trisfald) [inline README.md:47]: Do we perhaps want to override the warmup in `CI` to smaller value?

Comment (brunoborges) [reply README.md:47]: If we are going to have specific values in the CI workflow, how about we change the defaults in the shell script to values that we believe people should be using on a first shot at these benchmarks?

DURATION=300s
WARMUP=30s

For CI, we use:

DURATION=30s
WARMUP=15s

Wdyt?

Comment (LesnyRumcajs) [reply README.md:47]: @brunoborges This sounds good, I'd even decrease it further for the CI, e.g. 5s and 20/25s. The CI is only there to check if the benchmarks / services work at all, the Github Actions results are not reliable anyway.

Comment (brunoborges) [reply README.md:47]: Cool. I will add logic for the WARM UP, that if it is zero, it won't run.

I'll change the defaults to what we believe is best too, while adding low values for the CI workflow.

File: analyze.sh (hunk 1 above)
Comment (LesnyRumcajs) [inline analyze.sh:13]: It's already done in `bench.sh`, you could add the warmup param [there](https://github.com/LesnyRumcajs/grpc_bench/blob/master/bench.sh#L56)

File: ghz-tool/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
+FROM ubuntu
+
+RUN apt-get -q update \
+    && apt-get -q -y install wget \
+    && wget https://github.com/bojand/ghz/releases/download/v0.58.0/ghz-linux-x86_64.tar.gz \
```
Comment (LesnyRumcajs) [inline ghz-tool/Dockerfile:5]: Nice! Maybe we could use the latest stable release of `ghz`, unless the API drastically changed?

File: analyze.sh (hunk 1 above)
Comment (LesnyRumcajs) [reply analyze.sh:13]: @brunoborges sure, makes sense

**Pull request merged**